### PR TITLE
Download subregion

### LINKED ISSUE #44: Add support for regional subselection
From the [CDS documentation](https://confluence.ecmwf.int/display/CKB/How+to+download+ERA5):

> Currently the interactive CDS forms don't allow users to perform a regional sub-selection or an interpolation of the selected data. However, both actions can be performed using the following CDS API keywords:
For a geographical area subset, use key area.
For a different grid resolution, use key grid.
Please note that the ERA5 native grid in CDS is 0.25°x0.25° (atmosphere), 0.5°x0.5° (ocean waves), Mean, spread and members: 0.5°x0.5° (atmosphere), 1°x1° (ocean waves).

We want to integrate this option in era5cli.

### PR DESCRIPTION
**Pull request by aytacpacal**
Added ability to download subregion

closes #44

### CODE REVIEW

**Comment by Peter9192** [comment]
Hey @aytacpacal welcome and cheers for opening this PR! I added a link to a corresponding issue (#44).

File: era5cli/utils.py [hunk 1]
```diff
@@ -186,4 +186,4 @@ def _append_netcdf_history(ncfile: str, appendtxt: str):
             {}'''.format(appendtxt, ncfile.history))
     except AttributeError:
         ncfile.history = appendtxt
-    ncfile.close()
+    ncfile.close()
```
Comment (Peter9192) [inline era5cli/utils.py:189]: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Please use flake8 to see whether the code adheres to the PEP8 python style guidelines.

**Review by Peter9192 — CHANGES_REQUESTED** [review]
Hi @aytacpacal ,
Thank you very much for contribution this PR to era5cli! This is much appreciated. 

I have left a couple of comments and suggestions, to improve the code but mostly to make sure it is well documented. Additionally, I have the following suggestions:
- It would be good to add a check in the `cli.py` file and raise an error if invalid coordinates are passed.
- It would be good to also add a test for this new argument (see `tests/test_cli.py`)
- Please add your details to the `citation.cff` and `.zenodo.json` files if you would like to be recognized for your contributions.

Let me know if you need help with any of these recommendations. I'm happy to help.

**Comment by Peter9192** [comment]
Hello @aytacpacal ,
Thanks again for opening a pull request. I was wondering if you have seen my review, and whether you are planning to continue with this. Can you give an update?

**Comment by aytacpacal** [comment]
Hi @Peter9192, yes, thanks for your comments. Unfortunately, I did not have time to look. I plan to continue but do not know when it will be :(

**Comment by Peter9192** [comment]
> 
> 
> Hi @Peter9192, yes, thanks for your comments. Unfortunately, I did not have time to look. I plan to continue but do not know when it will be :(

Okay thanks for the reply. It's good to know this is still on your radar. Cheers!

**Comment by Peter9192** [comment]
Hello again @aytacpacal how are you doing? Can you give another update about your plans regarding this PR? If you don't have the time, we might be able to help out and finish the work on this one. Let me know what option you prefer :-)

**Comment by aytacpacal** [comment]
Hi @Peter9192, yeah I am struggling with my time management, so it will be better if someone else takes a look :(

**Comment by bvreede** [comment]
Hi @aytacpacal thanks again so much for opening this PR! 🎉  With your permission, I'll continue work on implementing the subregion download.

There are a few ways we can do this: as you may have noticed, I've copied your subregion branch to this repository, as I didn't have access to your forked repo. I am currently working on this branch.

Option 1: I can open a new PR with this branch.
Option 2: I can open a PR to your forked repository, which when it's merged would automatically update this current PR.

Working with a new PR (option 1) will be easier for me and wouldn't need your continuous attention, but while you would remain as the commit author for the commits already made on the branch, you wouldn't be the author of the PR itself. Should you want to add anything to the PR, option 2 is better, as you have no direct access to the branch on this repository.

Could you let me know which option you prefer?

Regarding recognition for your work: if you want, we can add your details to the citation.cff and .zenodo.json files, and I'm fairly certain you would be listed as a contributor on the github page as a commit author. And of course you will have our eternal gratitude ;)

Thanks again!

**Comment by aytacpacal** [comment]
Hi @bvreede, you can use whichever is easier for you :) I thank you for creating this software, your eternal gratitude is enough for me 🤗

**Comment by bvreede** [comment]
Hi @aytacpacal just to let you know, @Peter9192 just closed #70, and subregion support has been added! And you're now an official contributor to this repository :) Thanks much, all the best!

**Pull request closed without merging**